src/shared/parameter_v2/config_test.cpp

```diff
@@ -0,0 +1,264 @@
+#include <gtest/gtest.h>
+
+#include <boost/filesystem.hpp>
+#include <cstring>
+#include <regex>
+
+#include "clang-c/Index.h"
+#include "yaml-cpp/yaml.h"
+
+extern "C"
+{
+#include "shared/parameter_v2/c_parameters.h"
+}
+/**
+ * This function converts a config file names into config struct names.
+ *
+ * For example, a config file name maybe "example_config" gets converted
+ * to "ExampleConfig"
+ *
+ * @throws invalid_argument If input is empty or there is a _ at the end of the input
+ * @pre The snake_case_input must be valid snake_case
+ * @param The snake_case string to convert to CamelCase
+ * @return Converted string to CamelCase
```

Comment:
Nit: pre, param and return should have whitespace between them.

Reply:
added